examples/tagmap/app.js

```diff
@@ -0,0 +1,80 @@
+/* global window,document */
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+import MapGL from 'react-map-gl';
+import DeckGLOverlay from './deckgl-overlay.js';
+// handle ajax call
+import axios from 'axios';
+
+// Set your mapbox token here
+const MAPBOX_TOKEN = process.env.MapboxAccessToken; // eslint-disable-line
+// sample data
+const filePath = 'https://rivulet-zhang.github.io/dataRepo/tagmap/hashtags10k.json';
+
+class Root extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewport: {
+        ...DeckGLOverlay.defaultViewport,
+        width: 500,
+        height: 500
+      }
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this._resize.bind(this));
+    this._resize();
+    // set data in component state
+    this._loadData();
+  }
+
+  _resize() {
+    this._onViewportChange({
+      width: window.innerWidth,
+      height: window.innerHeight
+    });
+  }
+
+  _onViewportChange(viewport) {
+    this.setState({
+      viewport: {...this.state.viewport, ...viewport}
+    });
+  }
+
+  _loadData() {
+    // remove high-frequency terms
+    const excludeList = new Set(['#hiring', '#job', '#jobs', '#careerarc', '#career']);
+    const weightThreshold = 2;
+
+    axios.get(filePath)
```

Comment:
We usually use `d3-request` for http request since it's a lighter library. But no need to change it here, just mentioning for your info